From de09f9675e2db70fbd01daa47d97879c4d08e32a Mon Sep 17 00:00:00 2001
From: startergo <7897244+startergo@users.noreply.github.com>
Date: Wed, 24 Jun 2026 17:25:46 -0500
Subject: [PATCH 25/41] [sl-port-610] Fix scrollbar mouse-down crash:
 -[NSScrollerImp setTracking:] absent on 10.6 partial NSScrollerImp ->
 doesNotRecognizeSelector on scrollbar mouse-down. Guard with
 respondsToSelector: (companion to mouseEntered/ExitedScroller guards)

---
 Source/WebCore/platform/mac/ScrollAnimatorMac.mm | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/mac/ScrollAnimatorMac.mm b/Source/WebCore/platform/mac/ScrollAnimatorMac.mm
index aacc61e2b3..acc9b98940 100644
--- a/Source/WebCore/platform/mac/ScrollAnimatorMac.mm
+++ b/Source/WebCore/platform/mac/ScrollAnimatorMac.mm
@@ -1008,7 +1008,10 @@ void ScrollAnimatorMac::mouseIsDownInScrollbar(Scrollbar* scrollbar, bool mouseI
         return;
 
     if (NSScrollerImp *painter = scrollerImpForScrollbar(*scrollbar)) {
-        [painter setTracking:mouseIsDown];
+        // [leopard] -[NSScrollerImp setTracking:] is a later addition; absent on the 10.6
+        // partial NSScrollerImp -> doesNotRecognizeSelector crash on scrollbar mouse-down.
+        if ([painter respondsToSelector:@selector(setTracking:)])
+            [painter setTracking:mouseIsDown];
         if (mouseIsDown)
             [m_scrollerImpPair beginScrollGesture];
         else
-- 
2.54.0

